src/main/scala/org/bitcoins/core/util/Base58.scala

```diff
@@ -55,20 +64,132 @@ trait Base58 extends BitcoinSLogger {
+      case true => if (addressType == "pubkey") TestNet3.p2pkhNetworkByte else TestNet3.p2shNetworkByte
+      case false => if (addressType == "pubkey") MainNet.p2pkhNetworkByte else MainNet.p2shNetworkByte
+    }
+    val versionByte : Byte = parseVersionByte(addressType, isTestNet)
+    val bytes : Seq[Byte] = Seq(versionByte) ++ hash.bytes
+    val checksum = CryptoUtil.doubleSHA256(bytes).bytes.take(4)
+    encode(bytes ++ checksum)
+  }
+
+  /**
+    * Encodes a private key into Wallet Import Format (WIF)
+    * https://en.bitcoin.it/wiki/Wallet_import_format
+    *
+    * @param privateKey
+    * @param isCompressed
+    * @param isTestNet
+    * @return
+    */
+  //TODO: Create WIF PrivateKey Type
+  def encodePrivateKeyToWIF(privateKey : ECPrivateKey,
+                            isCompressed : Boolean,
+                            isTestNet : Boolean) : String = {
+    val versionByte : Byte = isTestNet match {
+      case true => TestNet3.privateKey
+      case false => MainNet.privateKey
+    }
+    val compressedByte : Option[Byte] = isCompressed match {
+      case true => Some(0x01.toByte)
+      case false => None
+    }
+    val bytes : Seq[Byte] = Seq(versionByte.toByte) ++ BitcoinSUtil.decodeHex(privateKey.hex) ++ compressedByte
+    val checksum =  CryptoUtil.doubleSHA256(bytes).bytes.take(4)
+    encode(bytes ++ checksum)
+  }
 
   /**
     * Takes in base58 string and returns sequence of bytes
     * https://github.com/ACINQ/bitcoin-lib/blob/master/src/main/scala/fr/acinq/bitcoin/Base58.scala
+    *
     * @param input base58 string to be decoded into a sequence of bytes
     * @return decoded sequence of bytes
     */
   def decode(input: String) : Seq[Byte] = {
     val zeroes = input.takeWhile(_ == '1').map(_ => 0:Byte).toArray
     val trim  = input.dropWhile(_ == '1').toList
-    val decoded = trim.foldLeft(BigInt(0))((a,b) =>a.*(BigInt(58L)).+(BigInt(base58Pairs(b))))
+    val decoded = trim.foldLeft(BigInt(0))((a,b) =>
+      a.*(BigInt(58L)).+(BigInt(base58Pairs(b))))
     if (trim.isEmpty) zeroes else zeroes ++ decoded.toByteArray.dropWhile(_ == 0)
   }
 
+  def isValid(base58 : String) : Boolean = {
+    if (base58.isEmpty) false
+    else {
+      val decoded = decode(base58)
+      val firstByte = decoded.head
+      val validAddressPreFixBytes: Seq[Byte] =
+        MainNetChainParams.base58Prefixes(PubKeyAddress) ++ MainNetChainParams.base58Prefixes(ScriptAddress) ++
+          TestNetChainParams.base58Prefixes(PubKeyAddress) ++ TestNetChainParams.base58Prefixes(ScriptAddress)
+      val validSecretKeyPreFixBytes : Seq[Byte] =
+        MainNetChainParams.base58Prefixes(SecretKey) ++ TestNetChainParams.base58Prefixes(SecretKey)
+      val compressedPubKey = List('K', 'L', 'c').contains(base58.head)
+      def checkCompressedPubKeyValidity : Boolean = {
+        val compressedByte = decoded(decoded.length - 5)
+        compressedByte == 0x01.toByte
+      }
+      try {
```

Comment:
Don't use `try`, use `Try` (with an uppercase `T`). This is better for various reasons, one of which is performance. You can use a match statement on the `Try` afterwards checking if it is a `Success` or `Failure`